crates/matrix-sdk/src/client/builder.rs

```diff
@@ -450,6 +450,7 @@ impl ClientBuilder {
 
         let http_client = HttpClient::new(inner_http_client.clone(), self.request_config);
 
+        #[allow(unused_variables)]
```

Comment:
This is moved up from the old `if let` (`well_known` is only used when `feature = "experimental-sliding-sync"` and Clippy wants to rename it).